src/wp-includes/i18n/class-wp-i18n-translations.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * I18N: WP_I18n_Translations class.
+ *
+ * @package WordPress
+ * @subpackage WP_I18n_Translation_Controller
+ * @since 6.5.0
+ */
+
+/**
+ * Class WP_I18n_Translations.
+ *
+ * @since 6.5.0
+ *
+ * @property-read array<string, string> $headers
+ * @property-read array<string, string[]> $entries
+ */
+class WP_I18n_Translations {
+	/**
+	 * Text domain.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var string
+	 */
+	protected $textdomain = 'default';
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $textdomain Text domain.
+	 */
+	public function __construct( string $textdomain = 'default' ) {
+		$this->textdomain = $textdomain;
+	}
+
+	/**
+	 * Magic getter for backward compatibility.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $name Property name.
+	 * @return mixed
+	 */
+	public function __get( string $name ) {
+		if ( 'entries' === $name ) {
+			$entries = WP_I18n_Translation_Controller::instance()->get_entries( $this->textdomain );
```

Comment:
Thoughts on passing an instance of WP_I18n_Translation_Controller via the constructor to the class so you can just use $this->controller->… (read dependency injection)

Reply:
We could do it, but isn't that a bit pointless when `WP_I18n_Translation_Controller` is a singleton?

Reply:
It makes this class more testable, so +1 to injecting via constructor.

Reply:
Well, here's a PR implementing your suggestion: https://github.com/swissspidy/wordpress-develop/pull/49

Is that what you had in mind?

Reply:
From only looking at the changes, yes! I know that core loves singletons, but that shouldn't prevent us from writing new code in a more modern way 🙃